h2/src/main/org/h2/store/fs/disk/FilePathDisk.java

```diff
@@ -55,23 +61,40 @@ public FilePathDisk getPath(String path) {
         return p;
     }
 
+    private FileSystem initFileSystem(URL url) throws Exception {
+        try {
+            return FileSystems.getFileSystem(url.toURI());
+        } catch (FileSystemNotFoundException e) {
+            Map<String, String> env = new HashMap<>();
+            env.put("create", "true");
+            return FileSystems.newFileSystem(url.toURI(), env);
+        }
+    }
+
     @Override
     public long size() {
         if (name.startsWith(CLASSPATH_PREFIX)) {
-            try {
-                String fileName = name.substring(CLASSPATH_PREFIX.length());
-                // Force absolute resolution in Class.getResource
-                if (!fileName.startsWith("/")) {
-                    fileName = "/" + fileName;
+            String path = this.name.substring("classpath:".length());
+            if (!path.startsWith("/")) {
+                path = "/" + path;
+            }
+            URL url = this.getClass().getResource(path);
+            if ("file".equals(url.getProtocol())) {
+                try {
+                    return Files.size(Paths.get(url.toURI()));
+                } catch (Exception e) {
+                    return 0L;
                 }
-                URL resource = this.getClass().getResource(fileName);
-                if (resource != null) {
-                    return Files.size(Paths.get(resource.toURI()));
+            }
+            try (FileSystem fs = initFileSystem(url)) {
```

Comment:
This method returns either an existing or a new file system.

Why existing file systems are closed here?

Reply:
Fixed